pbx/pbx_config.c

```diff
@@ -1994,6 +1994,9 @@ static void pbx_load_users(void)
 	if (!cfg)
 		return;
 
+	/*! \todo Remove users.conf support in Asterisk 23 */
+	ast_log(LOG_WARNING, "users.conf is deprecated and will be removed in a future version of Asterisk\n");
```

Comment:
I think this should use the same approach in other places and subscribe to know that Asterisk is fully booted, at which point the users.conf warning can be output so it appears after all of the other logging that can occur.

Reply:
If we do this in other places as well, should it use a common subscription so that a single warning is emitted, or all separate? I'm just not sure how'd you do that without symbols being visible outside of individual files.

My other thought would be instead of every place where the config loads, would it be fine to put it in ast_config_load?

e.g.

`if (!strcmp(cfg_filename, "users.conf")) warning()`

That would cover allow a single warning to cover anything possibly using users.conf. On the downside, it would add a slight amount of overhead to every config load.